gensim/similarity_learning/evaluation_scripts/w2v_avg_eval.py

```diff
@@ -0,0 +1,144 @@
+import numpy as np
+import pickle
+import os
+from pprint import pprint
+import random
+
+from gensim.similarity_learning import WikiQAExtractor
+
+def accuracy(Y_true, Y_pred):
+    n_correct = 0
+    for y_pred, y_true in zip(Y_pred, Y_true):
+        if (np.argmax(y_true) == np.argmax(y_pred)):
+            n_correct += 1
+
+    return n_correct/len(Y_true)
+
+
+
+def my_map(Y_true, Y_pred):
+    average_precisions = []
+
+    for y_true, y_pred in zip(Y_true, Y_pred):
+
+        pred_sorted = sorted(zip(y_true, y_pred), key=lambda x:x[1], reverse=True)
+
+        avg = 0
+        n_relevant = 0
+
+        for i, val in enumerate(pred_sorted):
+            if val[0] == 1:
+                avg += 1./(i + 1.)
+                n_relevant += 1
+
+        if n_relevant != 0:
+            ap = avg/n_relevant
+            average_precisions.append(ap)
+            
+    return np.mean(np.array(average_precisions))
+
+
+def mean_ndcg(Y_true, Y_pred, k=10):
+    
+    ndcgs = []
+
+    for y_true, y_pred in zip(Y_true, Y_pred):
+
+        pred_sorted = sorted(zip(y_true, y_pred), key=lambda x:x[1], reverse=True)
+        true_sorted = sorted(zip(y_true, y_pred), key=lambda x:x[0], reverse=True)
+
+        # make cutoff for @k
+        pred_sorted = pred_sorted[:k]
+        true_sorted = true_sorted[:k]
+
+        dcg = 0
+
+        for i, val in enumerate(pred_sorted):
+            if val[0] == 1:
+                dcg += 1./np.log2(i + 2)
+
+
+        idcg = 0
+
+        for i, val in enumerate(true_sorted):
+            if val[0] == 1:
+                idcg += 1./np.log2(i + 2)
+
+        if idcg != 0:
+            ndcgs.append(dcg/idcg)
+
+    return np.mean(np.array(ndcgs))
+
+def sent2vec(w2v, sentence):
+    vec_sum = []
+
+    for word in sentence.split():
+        if word in w2v:
+            vec_sum.append(w2v[word])
+
+    return np.mean(np.array(vec_sum), axis=0)
+
+def cos_sim(vec1, vec2):
+    return np.sum(vec1*vec2)/(np.linalg.norm(vec1)*np.linalg.norm(vec2))
+
+
+if __name__ == '__main__':
+
+
+    w2v = {}
+
+    with open('glove.6B.50d.txt') as f:

```

Comment:
For this, you can use `gensim-data` https://github.com/RaRe-Technologies/gensim-data/releases